Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/DataStream/DataSource/Consolidator/EntityProxyActiveConsolidator.cs

```diff
@@ -0,0 +1,113 @@
+using Anvil.Unity.DOTS.Data;
+using System;
+using System.Diagnostics;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    [BurstCompatible]
+    internal unsafe struct EntityProxyActiveConsolidator<TInstance>
+        where TInstance : unmanaged, IEntityProxyInstance
+    {
+        private static readonly int ELEMENT_SIZE = sizeof(EntityProxyInstanceWrapper<TInstance>);
+
+
+        private UnsafeParallelHashMap<EntityProxyInstanceID, bool> m_CancelRequestsLookup;
+        [NativeDisableUnsafePtrRestriction] private readonly void* m_ActiveBufferPointer;
+        [NativeDisableUnsafePtrRestriction] private readonly void* m_PendingCancelActiveBufferPointer;
+        private readonly uint m_PendingCancelActiveID;
+        private readonly CancelBehaviour m_CancelBehaviour;
+
+
+        public EntityProxyActiveConsolidator(ActiveArrayData<EntityProxyInstanceWrapper<TInstance>> activeArrayData) : this()
+        {
+            m_ActiveBufferPointer = activeArrayData.Active.GetBufferPointer();
+            Debug_EnsurePointerNotNull(m_ActiveBufferPointer);
+
+            if (activeArrayData.PendingCancelActiveData != null)
+            {
+                ActiveArrayData<EntityProxyInstanceWrapper<TInstance>> pendingCancelActiveData = (ActiveArrayData<EntityProxyInstanceWrapper<TInstance>>)activeArrayData.PendingCancelActiveData;
+                m_PendingCancelActiveBufferPointer = pendingCancelActiveData.Active.GetBufferPointer();
+                Debug_EnsurePointerNotNull(m_PendingCancelActiveBufferPointer);
+
+                m_PendingCancelActiveID = pendingCancelActiveData.ID;
+            }
+            
+            m_CancelBehaviour = activeArrayData.CancelBehaviour;
+            m_CancelRequestsLookup = activeArrayData.TaskSetOwner.TaskSet.CancelRequestsDataStream.ActiveLookupData.Lookup;
```

Comment:
you getting this lookup
![so-close-reaching-out](https://user-images.githubusercontent.com/1426795/213747102-2a29741b-ba91-4416-a94e-bcd3411c2de2.gif)

Reply:
Lol! Hopefully #136, #137 or #138 will clean a bunch of this stuff up.